File: DATABASE-MANAGEMENT/addApplications.cpp
/* Compile using:
g++ -Wall -I/usr/include/cppconn -o odbc addStudents.cpp -L/usr/lib -lmysqlcppconn
run:
./odbc */
#include "odbc_db.h"
#include <iostream>
#include <string>
using namespace std;

int main(int argc, char *argv[])
{
string Username = "bkj011";   // Change to your own username
string mysqlPassword = "oa8ahYoo";  // Change to your mysql password
string SchemaName = "bkj011"; // Change to your username

   odbc_db myDB;
   myDB.Connect(Username, mysqlPassword, SchemaName);
   myDB.initDatabase();

   // For debugging purposes:  Show the database before insert
   string builder = "";
   string query1 = "SELECT * from APPLICATIONS;";
   //builder.append("<br><br><br> APPLICATIONS table before:" + myDB.query(query1) +"<br>");

   // Parse input string to get job information
   string studentID;
   string jobID;

   // Read command line arguments
   // First arg, arg[0] is the name of the program
   // Next args are the parameters
   studentID = argv[1];
   jobID = argv[2];
   
   if(studentID.compare("") == 0 || jobID.compare("") == 0){
	   cout << "INVALID INPUT" << endl;
	   return 0;
   }

   // Insert the new student
   string input = "'" + studentID + "','" + jobID + "'";

   // DEBUG:
   // printf("%s", input.c_str());
   myDB.insert("APPLICATIONS", input);    // insert new student

   //For debugging purposes: Show the database after insert
   builder.append("<br><br><br> Table APPLICATIONS after:" + myDB.query(query1));
   cout << builder;

   myDB.disConnect();//disconect Database

   return 0;
}

File: DATABASE-MANAGEMENT/addApplications.php
<html>
<body>
<center>
<div style="width:800px;height:1000px;border:10px solid #151973;background-color:rgba(255,255,255,0.84)">
<body background="https://img.freepik.com/free-vector/circuit-board-seamless-pattern_98292-3920.jpg">
<h3>Enter information about an application to add to the database:</h3>

<div>
    <b>Example Applications: </b>
    <table border = "2" bordercolor="330033">
    <thead>
    <tr>
        <th>Student ID</th>
	<th>Job ID</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td>0</td>
	<td>1000</td>
    </tr>
    <tr>
        <td>1</td>
	<td>1001</td>
    </tr>
    </tbody>
    </table>
</div>
<br>
<?php
include("php_db.php");
$myDb = new php_db('bkj011', 'oa8ahYoo', 'bkj011');

//Init database
$myDb->initDatabase();

$Students = $myDb->query('SELECT * FROM STUDENTS');
$Jobs = $myDb->query('SELECT * FROM JOBS');
?>
<form action="addApplications.php" method="post">
	<a>Student: </a>
	<select name="StudentChoice">
		<option value="">--- Select ---</option>
		<?php foreach ($Students as $row) { ?>
		<option value="<?php echo $row['STUDENTID'] ?>"><?php echo $row['STUDENTID'] . ': ' . $row['STUDENTNAME'] ?></option>
		<?php
		}
		?>
	</select>
	<br>
	<br>
	<a>Jobs: </a>
	<select name="JobChoice">
		<option value="">--- Select ---</option>
		<?php foreach ($Jobs as $row) { ?>
		<option value="<?php echo $row['JOBID'] ?>"><?php echo $row['JOBID'] . ': ' . $row['JOBTITLE'] ?></option>
		<?php
		}
		?>
	</select>
	<br>
	<br>
	<input name="submit" type="submit">
</form>

<?php
$myDb->disconnect();
?>
<br><br>

<a href="http://www.csce.uark.edu/~bkj011/project_cpp/hub.html">Go Back to Hub Page!</a>

<?php
if (isset($_POST['submit'])) 
{
    // replace ' ' with '\ ' in the strings so they are treated as single command line args
    $studentID = escapeshellarg($_POST[StudentChoice]);
    $jobID = escapeshellarg($_POST[JobChoice]);

    $command = '/home/bkj011/public_html/project_cpp/addApplications.exe ' . $studentID . ' ' . $jobID;

    //echo '<p>' . 'command: ' . $command . '<p>';
    // remove dangerous characters from command to protect web server
    $command = escapeshellcmd($command);
 
    // run addStudents.exe
    system($command);           
}
?>
</body>
</div>
</center>
</body>
</html>
